executor/adapter.go

```diff
@@ -289,7 +289,11 @@ func (a *ExecStmt) buildExecutor(ctx sessionctx.Context) (Executor, error) {
 			err = ctx.InitTxnWithStartTS(math.MaxUint64)
 		} else {
 			log.Debugf("con:%d ActivePendingTxn %s", ctx.GetSessionVars().ConnectionID, a.Text)
-			err = ctx.ActivePendingTxn()
+			if _, ok := a.Plan.(*plannercore.CheckTable); ok && ctx.GetSessionVars().SnapshotTS != 0 {
```

Comment:
`ctx.GetSessionVars().SnapshotTS != 0` should be firstly check. For the performance.